pybop/plotting/plot_convergence.py

```diff
@@ -29,7 +30,7 @@ def plot_convergence(
 
     # Compute the minimum cost for each iteration
     min_cost_per_iteration = [
-        min(cost_function(solution) for solution in log_entry)
+        min((cost_function(solution) for solution in log_entry), default=np.inf)
```

Comment:
We should just save the cost value during optimisation and plot it here. At the moment, this plot takes quite a while since it completes the optimisation a second time. 

Does the `solution` object contain the cost value? We can push this to a future PR, but should probably open an issue if we don't fix it here.

Reply:
This edit is simply to cope with failures in the evaluation of the cost function which, in unusual cases, left an empty array with no minimum value.

Reply:
I've copied your comment to the plotting PR: https://github.com/pybop-team/PyBOP/pull/178/files#r1507377660